google-cloud-clients/google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Publisher.java

```diff
@@ -430,7 +429,32 @@ public void shutdown() throws Exception {
    * <p>Call this method to make sure all resources are freed properly.
    */
   public boolean awaitTermination(long duration, TimeUnit unit) throws InterruptedException {
-    return publisherStub.awaitTermination(duration, unit);
+    long startDuration = System.currentTimeMillis();
```

Comment:
please make `startDuration` final.  There's never a need to update it, since startDuration represents the start time when the method was called.

Reply:
@sduskis startDuration needs to be update because it is used(counting) to find elapsed time which occur to execute waitNoMessages method only, so we have to minus this time from actual time pass to  duration parameter of awaitTermination method and so on needs to update every time to find elapsed time so we can remove it from total remaining time.

Reply:
I think I understand what you mean.  Let's have a static `startTime` and `totalDurationMs` variables that are final.  We can use those plus current time to calculate `remainingDurationMs`

Reply:
@sduskis chnages done